test/queue_list_test.dart

```diff
@@ -258,6 +258,25 @@ void main() {
     expect(queue.cast<Pattern>(), same(queue));
   }, skip: isDart2 ? false : 'Requires a Dart2 runtime');
 
+  test("cast does not throw on mutation when the type is valid", () {
+    var patternQueue = new QueueList<Pattern>()..addAll(['a', 'b']);
+    QueueList<Object> untypedQueue = patternQueue;
+    var stringQueue = untypedQueue.cast<String>();
+    stringQueue.addAll(['c', 'd']);
+
+    expect(stringQueue, ['a', 'b', 'c', 'd']);
+  });
+
+  test("cast throws on mutation when the type is not valid", () {
+    QueueList<Object> stringQueue = new QueueList<String>();
+    var numQueue = stringQueue.cast<num>();
+    expect(
+      () => numQueue.add(1),
+      throwsCastError,
+      skip: isDart2 ? false : 'In checked mode (VM), CastError is not thrown',
```

Comment:
The real reason is that `cast` only works in Dart 2 because it depends on the type parameter of a generic method. It's not VM specific in any way.

(In checked mode, the `add(1)` *would* throw because you are adding an `int` to a `QueueList<String>`, but it's more likely to be a `TypeError` than a `CastError`).

Reply:
Thanks. Done!

Reply:
Looks like the type thrown is `CastErrorImplementation`, which is not a `TypeError`.